collect/collect.go

```diff
@@ -196,68 +195,46 @@ func (i *InMemCollector) AddSpanFromPeer(sp *types.Span) {
 // block is the only place we are allowed to modify any running data
 // structures.
 func (i *InMemCollector) collect() {
-	// make sure we send all pending traces when we finish
-	defer func() {
-		for sig := range i.toSend {
-			i.send(sig.trace)
-		}
-	}()
-
-	peerChanSize := cap(i.fromPeer)
+	ticker := time.NewTicker(i.Config.GetSendTickerValue())
+	defer ticker.Stop()
+
+	incoming := mergeIncomingSpans(spanInput{
+		ch:          i.incoming,
+		name:        "from_incoming",
+		concurrency: 1,
+	}, spanInput{
+		ch:          i.fromPeer,
+		name:        "from_peer",
+		concurrency: 2,
+	})
 
 	for {
 		// record channel lengths
-		i.Metrics.Histogram("collector_tosend_queue", float64(len(i.toSend)))
-		i.Metrics.Histogram("collector_incoming_queue", float64(len(i.incoming)))
```

Comment:
Is there any way to still record these channel lengths before we merge them? Seems like that could be helpful, but mostly asking out of curiosity.